service/src/client.rs

```diff
@@ -263,3 +264,28 @@ fn convert_metadata(meta: PostMetadata) -> spacemesh_v1::Metadata {
         labels_per_unit: meta.labels_per_unit,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use tonic::transport::{Certificate, Identity};
+
+    #[test]
+    fn derives_domain_from_address() {
+        let crt = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
+        let client_crt = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
+        super::ServiceClient::new(
+            "https://localhost:1234".to_string(),
```

Comment:
Does this also work with

```suggestion
            "https://127.0.0.1:1234".to_string(),
```

Reply:
Yes